adhocteam/soapbox
# Add user_data script for app instances

For #28. What's included:

- A `user_data.tmpl` template. Value of the rendered template is used as `user_data` for an app's launch config.
- A few tweaks to the ALB configuration to support passing an `application_port` to the `user_data` script.
- Modifications to the networking configs to allow egress from app servers and make `ssh` access to app servers possible (have to use a bastion in `dmz` subnet for now).

How to test this:

Note: this example will use `example-web-app.tar.gz` produced from testing #44 and the AMI from testing #45.

1. Create the example app's VPC:

```
cd ops/terraform/aws/network
$ terraform apply \
  -var 'application_domain=example-web-app.com' \
  -var 'application_name=example-web-app' \
  -var 'environment=prod'
```

2. Set up the example app's deployment config:

```
cd ../deployment/asg
$ terraform get
$ terraform apply \
  -var "application_name=example-web-app" \
  -var "environment=prod" \
  -var "health_check_path=/_healthcheck"
  -var "release=1"
```

At this point, you should be able to view the example app using the ALB's hostname.
